test/src/Makefile.am

```diff
@@ -1,9 +1,17 @@
+AUTOMAKE_OPTIONS        = subdir-objects
+
 # Add source of new files to be in tarball here
-EXTRA_DIST              = unittest.h xht.c
+EXTRA_DIST              = unittest.h xht.c addr_test.c
 
 # Add new test programs here
-noinst_PROGRAMS         = xht #new-test
+noinst_PROGRAMS         = xht addr
+
+addr_SOURCES            = addr_test.c
+
```

Comment:
This needs to be `noinst_PROGRAMS = xht addr_test` and `addr_test_SOURCES = ...`

Reply:
Well, that was on purpose.
I chose to name the test source file different from the normal source file, because a) I include the normal source file in the test and that can get collisions depending where the object file is placed, and b) because I thought that having a different name could be clearer.
But I did leave the binary name as `addr`, following your example, and since there would be no collisions and it will keep the lines shorter and in general looks good.

So which route should we go? Keep it consistent and name ne binary `addr_test` if the source file is called `addr_test`? I am file with that. I just see that there might not always be a 1-to-1 correlation as a test could be made up of multiple source files.

Reply:
Ah sorry, now I see.  I was on the train talking with a colleague at the same time.

Go ahead with your approach, we can always rename and restructure later, i.e., I'm fine with merging the PR as it is now.